client/src/main/java/org/asynchttpclient/cookie/ThreadSafeCookieStore.java

```diff
@@ -126,13 +149,6 @@ private boolean hasCookieExpired(Cookie cookie, long whenCreated) {
       return false;
   }
 
-  // rfc6265#section-5.1.3
-  // check "The string is a host name (i.e., not an IP address)" ignored
-  private boolean domainsMatch(String cookieDomain, String requestDomain, boolean hostOnly) {
-    return (hostOnly && Objects.equals(requestDomain, cookieDomain)) ||
-            (Objects.equals(requestDomain, cookieDomain) || requestDomain.endsWith("." + cookieDomain));
-  }
-
```

Comment:
It looks to me that your whole PR is built on the assumption matching domains on strict equality is enough.
This is wrong, please read RFC6265, in particular section about domain matching.